packages/components/src/internal/components/gridbar/GridAliquotViewSelector.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2019 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { Component, ReactNode } from 'react';
+import { Checkbox, DropdownButton, MenuItem } from 'react-bootstrap';
+import { List } from 'immutable';
+
+import { Filter } from "@labkey/api";
+
+import { generateId, QueryGridModel, QueryModel } from '../../..';
+import { getCheckedValue } from "../domainproperties/actions";
+import { replaceFilter } from "../../util/URL";
+
+interface Props {
+    queryGridModel?: QueryGridModel;
+    queryModel?: QueryModel;
+    updateFilter?: (filter: Filter.IFilter, filterColumnToRemove?: string) => void
+}
+
+enum MODE {
+    all,
+    samples,
+    aliquots,
+    none//when using omni filter with 'is blank'
+}
+
+const IS_ALIQUOT_COL = "IsAliquot";
+
+//TODO add jest tests
+export class GridAliquotViewSelector extends Component<Props> {
+    dropId: string;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.dropId = generateId('aliquotviewselector-');
+    }
+
+    handleCheckboxChange = evt => {
+        const check = getCheckedValue(evt);
+        const isAliquot = evt.target.id === 'checkbox-aliquot';
+
+        this.updateAliquotFilter(isAliquot, check);
+    };
+
+    updateAliquotFilter(isAliquot: boolean, check: boolean) {
+        const { queryGridModel, queryModel, updateFilter } = this.props;
+        const filterMode = this.getAliquotFilterMode();
+
+        let newFilter;
+        if (filterMode == MODE.all || filterMode == MODE.none)
+            newFilter = Filter.create(IS_ALIQUOT_COL, (isAliquot && check) || (!isAliquot && !check));
+        else
+            newFilter = null; // if neither is checked, or if both are checked, clear the filter
+
+        if (queryGridModel) {
+            replaceFilter(queryGridModel, IS_ALIQUOT_COL, newFilter);
+        }
+        else if (updateFilter) {
+            updateFilter(newFilter, IS_ALIQUOT_COL);
+        }
+
+    };
+
+    getTitle(mode: MODE) {
+        switch (mode) {
+            case MODE.samples:
+                return 'Samples only';
+            case MODE.aliquots:
+                return 'Aliquots only';
+            case MODE.none:
+                return 'None';
+            default:
+                return 'All samples amount';
```

Comment:
Should this be just "All samples"?  I know this is what's in the spec, but the wording is odd to me.  We aren't using this menu to change any of the "stored amounts" values, right?  it just determines which subset of samples to show in the grid.  "Amount" just seems confusing.